### PR TITLE
Fixed NPE getting Stack Trace if Throwable is null

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java [hunk 1]
```diff
@@ -48,6 +48,12 @@ public void testContextedException() {
         assertTrue(StringUtils.isEmpty(message));
     }
 
+    @Test
+    public void testNullException() {
+        final String trace = ExceptionUtils.getStackTrace(null);
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java:53]: In-line `trace`.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java]: Hi @garydgregory 
Changed
TY

File: src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java [hunk 2]
```diff
@@ -48,6 +48,11 @@ public void testContextedException() {
         assertTrue(StringUtils.isEmpty(message));
     }
 
+    @Test
+    public void testNullException() {
+        assertTrue(StringUtils.isEmpty(ExceptionUtils.getStackTrace(null)));
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java:53]: I think the test can be simplified by not using StringUtils.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java]: HI @garydgregory 
SomeThins like --> `assertNotNull(ExceptionUtils.getStackTrace(null));` ??
TY

Comment (garydgregory) [reply src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java]: > 
> 
> HI @garydgregory
> SomeThins like --> `assertNotNull(ExceptionUtils.getStackTrace(null));` ??
> TY

Hello @arturobernalg 
No, assert what you expect to happen on null input. Your suggestion above would accept a return value of "foo". Is that OK? I think not.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/exception/ContextedExceptionTest.java]: HI @garydgregory 
ok i think i understand what you mean.
TY

File: src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java [hunk 3]
```diff
@@ -359,14 +359,16 @@ public static String getRootCauseMessage(final Throwable th) {
...
-     *  {@code printStackTrace(PrintWriter)} method
+     * {@code printStackTrace(PrintWriter)} method, or an empty String if {@code null} input
      */
     public static String getStackTrace(final Throwable throwable) {
         final StringWriter sw = new StringWriter();
         final PrintWriter pw = new PrintWriter(sw, true);
-        throwable.printStackTrace(pw);
+        if (throwable != null) {
+            throwable.printStackTrace(pw);
+        }
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java:371]: We do not need to create the writers at all. IMO, just guard the method by starting it with:
```
        if (throwable == null) {
            return StringUtils.EMPTY;
        }
```

**Review by garydgregory — CHANGES_REQUESTED** [review]
@arturobernalg 
See my 1 comment and rebase on master, please.

**Review by garydgregory — CHANGES_REQUESTED** [review]
See https://github.com/apache/commons-lang/pull/733/files#r820237567

File: src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java (hunk 3 above)
Comment (garydgregory) [reply src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java:378]: ping.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java:378]: HI @garydgregory 
I don't see the way to just use String and not break the binary.

**Pull request merged**